contrib/kythe/src/python/pants/contrib/kythe/tasks/index_java.py

```diff
@@ -0,0 +1,85 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from pants.backend.jvm.subsystems.jvm import JVM
+from pants.backend.jvm.tasks.nailgun_task import NailgunTask
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.java.jar.jar_dependency import JarDependency
+from pants.util.dirutil import safe_mkdir
+
+from pants.contrib.kythe.tasks.indexable_java_targets import IndexableJavaTargets
+
+
+class IndexJava(NailgunTask):
+  _KYTHE_INDEXER_MAIN = 'com.google.devtools.kythe.analyzers.java.JavaIndexer'
+
+  cache_target_dirs = True
+
+  @classmethod
+  def implementation_version(cls):
+    # Bump this version to invalidate all past artifacts generated by this task.
+    return super(IndexJava, cls).implementation_version() + [('IndexJava', 6), ]
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(IndexJava, cls).subsystem_dependencies() + (JVM.scoped(cls),)
```

Comment:
@benjyw : AFAICT, you're grabbing the JVM options explicitly from the subsystem, but most nailgun tasks use `self.context.get_options().jvm_options`, based on the assumption that they're provided (where that is better or worse is... unclear). So you needn't directly reference the `JVM` subsystem here.